"invalid memory address or nil pointer dereference" when completing a multipart copy.

## Expected Behavior
It should be possible to do a multipart copy of a large object stored in Minio with no errors being thrown.

## Current Behavior
When completing the multipart copy, the copy fails with:
```
Caused by: com.amazonaws.services.s3.model.AmazonS3Exception: The specified multipart upload does not exist. The upload ID may be invalid, or the upload may have been aborted or completed. (Service: Amazon S3; Status Code: 404; Error Code: NoSuchUpload; Request ID: 1551CC19016E9F8B), S3 Extended Request ID: 3L137
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.handleErrorResponse(AmazonHttpClient.java:1545)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeOneRequest(AmazonHttpClient.java:1183)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeHelper(AmazonHttpClient.java:964)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.doExecute(AmazonHttpClient.java:676)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeWithTimer(AmazonHttpClient.java:650)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.execute(AmazonHttpClient.java:633)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.access$300(AmazonHttpClient.java:601)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutionBuilderImpl.execute(AmazonHttpClient.java:583)
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:447)
	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:4031)
	at com.amazonaws.services.s3.AmazonS3Client.completeMultipartUpload(AmazonS3Client.java:2804)
	at com.amazonaws.services.s3.transfer.internal.CompleteMultipartCopy.call(CompleteMultipartCopy.java:83)
	at com.amazonaws.services.s3.transfer.internal.CompleteMultipartCopy.call(CompleteMultipartCopy.java:38)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

Minio displays this panic in its log.
```
http: panic serving 127.0.0.1:35162: runtime error: invalid memory address or nil pointer dereference
goroutine 108 [running]:
net/http.(*conn).serve.func1(0xc4202da320)
	/opt/go/src/net/http/server.go:1726 +0xd0
panic(0x12788a0, 0x202b760)
	/opt/go/src/runtime/panic.go:502 +0x229
github.com/minio/minio/cmd.criticalErrorHandler.ServeHTTP.func1(0x15910c0, 0xc4204207e0, 0xc42066c100)
	/q/.q/sources/gopath/src/github.com/minio/minio/cmd/generic-handlers.go:771 +0x93
panic(0x12788a0, 0x202b760)
	/opt/go/src/runtime/panic.go:502 +0x229
github.com/minio/minio/cmd.(*FSObjects).CompleteMultipartUpload(0xc4201e3730, 0x1592640, 0xc42047c120, 0xc42003e056, 0x4, 0xc42003e05b, 0x3, 0xc42003e068, 0x24, 0xc4205a2000, ...)
	/q/.q/sources/gopath/src/github.com/minio/minio/cmd/fs-v1-multipart.go:590 +0x1beb
github.com/minio/minio/cmd.(ObjectLayer).CompleteMultipartUpload-fm(0x1592640, 0xc42047c120, 0xc42003e056, 0x4, 0xc42003e05b, 0x3, 0xc42003e068, 0x24, 0xc4205a2000, 0xcd, ...)
	/q/.q/sources/gopath/src/github.com/minio/minio/cmd/object-handlers.go:1702 +0x114
github.com/minio/minio/cmd.objectAPIHandlers.CompleteMultipartUploadHandler(0x14be408, 0x14be400, 0x158b640, 0xc42055e180, 0xc420346400)
	/q/.q/sources/gopath/src/github.com/minio/minio/cmd/object-handlers.go:1706 +0x78f
github.com/minio/minio/cmd.(objectAPIHandlers).CompleteMultipartUploadHandler-fm(0x158b640, 0xc42055e180, 0xc420346400)
	/q/.q/sources/gopath/src/github.com/minio/minio/cmd/api-router.go:58 +0x57
net/http.HandlerFunc.ServeHTTP(0xc42048d0e0, 0x158b640, 0xc42055e180, 0xc420346400)
	/opt/go/src/net/http/server.go:1947 +0x44
github.com/minio/minio/vendor/github.com/gorilla/mux.(*Router).ServeHTTP(0xc420522140, 0x158b640, 0xc42055e180, 0xc420346400)
	/q/.q/sources/gopath/src/github.com/minio/minio/vendor/github.com/gorilla/mux/mux.go:107 +0xdc
github.com/minio/minio/cmd.requestIDHeaderHandler.ServeHTTP(0x1580e40, 0xc420522140, 0x158b640, 0xc42055e180, 0xc42066c100)
	/q/.q/sources/gopath/src/github.com/minio/minio/cmd/generic-handlers.go:742 +0x9e
github.com/minio/minio/cmd.securityHeaderHandler.ServeHTTP(0x1582fa0, 0xc420543630, 0x158b640, 0xc42055e180, 0xc42066c100)
	/q/.q/sources/gopath/src/github.com/minio/minio/cmd/generic-handlers.go:757 +0xdc
github.com/minio/minio/cmd.bucketForwardingHandler.ServeHTTP(0xc420591020, 0x1583020, 0xc420543640, 0x158b640, 0xc42055e180, 0xc42066c100)
	/q/.q/sources/gopath/src/github.com/minio/minio/cmd/generic-handlers.go:628 +0x88
github.com/minio/minio/cmd.rateLimit.ServeHTTP(0xc420038c30, 0x1582e20, 0xc420591040, 0x158b640, 0xc42055e180, 0xc42066c100)
	/q/.q/sources/gopath/src/github.com/minio/minio/cmd/generic-handlers.go:722 +0x123
github.com/minio/minio/cmd.pathValidityHandler.ServeHTTP(0x1582f40, 0xc420591060, 0x158b640, 0xc42055e180, 0xc42066c100)
	/q/.q/sources/gopath/src/github.com/minio/minio/cmd/generic-handlers.go:615 +0x1b8
github.com/minio/minio/cmd.httpStatsHandler.ServeHTTP(0x1582f20, 0xc420543660, 0x15910c0, 0xc4204207e0, 0xc42066c100)
	/q/.q/sources/gopath/src/github.com/minio/minio/cmd/generic-handlers.go:554 +0xba
github.com/minio/minio/cmd.requestSizeLimitHandler.ServeHTTP(0x1582ec0, 0xc420543670, 0x50004000000, 0x15910c0, 0xc4204207e0, 0xc42066c100)
	/q/.q/sources/gopath/src/github.com/minio/minio/cmd/generic-handlers.go:71 +0xde
github.com/minio/minio/cmd.requestHeaderSizeLimitHandler.ServeHTTP(0x1582fc0, 0xc420591080, 0x15910c0, 0xc4204207e0, 0xc42066c100)
	/q/.q/sources/gopath/src/github.com/minio/minio/cmd/generic-handlers.go:96 +0xa7
github.com/minio/minio/cmd.crossDomainPolicy.ServeHTTP(0x1582f80, 0xc420543680, 0x15910c0, 0xc4204207e0, 0xc42066c100)
	/q/.q/sources/gopath/src/github.com/minio/minio/cmd/crossdomain-xml-handler.go:51 +0x8a
github.com/minio/minio/cmd.redirectHandler.ServeHTTP(0x1582e80, 0xc420543690, 0x15910c0, 0xc4204207e0, 0xc42066c100)
	/q/.q/sources/gopath/src/github.com/minio/minio/cmd/generic-handlers.go:248 +0x92
github.com/minio/minio/cmd.minioReservedBucketHandler.ServeHTTP(0x1582f60, 0xc4205436a0, 0x15910c0, 0xc4204207e0, 0xc42066c100)
	/q/.q/sources/gopath/src/github.com/minio/minio/cmd/generic-handlers.go:306 +0x78
github.com/minio/minio/cmd.cacheControlHandler.ServeHTTP(0x1582f00, 0xc4205436b0, 0x15910c0, 0xc4204207e0, 0xc42066c100)
	/q/.q/sources/gopath/src/github.com/minio/minio/cmd/generic-handlers.go:275 +0x73
github.com/minio/minio/cmd.timeValidityHandler.ServeHTTP(0x1582e40, 0xc4205436c0, 0x15910c0, 0xc4204207e0, 0xc42066c100)
	/q/.q/sources/gopath/src/github.com/minio/minio/cmd/generic-handlers.go:376 +0x1af
github.com/minio/minio/vendor/github.com/rs/cors.(*Cors).Handler.func1(0x15910c0, 0xc4204207e0, 0xc42066c100)
	/q/.q/sources/gopath/src/github.com/minio/minio/vendor/github.com/rs/cors/cors.go:190 +0xfe
net/http.HandlerFunc.ServeHTTP(0xc420591120, 0x15910c0, 0xc4204207e0, 0xc42066c100)
	/opt/go/src/net/http/server.go:1947 +0x44
github.com/minio/minio/cmd.resourceHandler.ServeHTTP(0x1583fa0, 0xc420591120, 0x15910c0, 0xc4204207e0, 0xc42066c100)
	/q/.q/sources/gopath/src/github.com/minio/minio/cmd/generic-handlers.go:505 +0xd2
github.com/minio/minio/cmd.authHandler.ServeHTTP(0x1583000, 0xc420543730, 0x15910c0, 0xc4204207e0, 0xc42066c100)
	/q/.q/sources/gopath/src/github.com/minio/minio/cmd/auth-handler.go:287 +0x157
github.com/minio/minio/cmd.reservedMetadataHandler.ServeHTTP(0x1582e00, 0xc420543740, 0x15910c0, 0xc4204207e0, 0xc42066c100)
	/q/.q/sources/gopath/src/github.com/minio/minio/cmd/generic-handlers.go:139 +0xa7
github.com/minio/minio/cmd.criticalErrorHandler.ServeHTTP(0x1582fe0, 0xc420543750, 0x15910c0, 0xc4204207e0, 0xc42066c100)
	/q/.q/sources/gopath/src/github.com/minio/minio/cmd/generic-handlers.go:774 +0x8d
github.com/minio/minio/cmd/http.(*Server).Start.func1(0x15910c0, 0xc4204207e0, 0xc42066c100)
	/q/.q/sources/gopath/src/github.com/minio/minio/cmd/http/server.go:115 +0xe7
net/http.HandlerFunc.ServeHTTP(0xc42055e080, 0x15910c0, 0xc4204207e0, 0xc42066c100)
	/opt/go/src/net/http/server.go:1947 +0x44
net/http.serverHandler.ServeHTTP(0xc420122900, 0x15910c0, 0xc4204207e0, 0xc42066c100)
	/opt/go/src/net/http/server.go:2694 +0xbc
net/http.(*conn).serve(0xc4202da320, 0x1592580, 0xc420676000)
	/opt/go/src/net/http/server.go:1830 +0x651
created by net/http.(*Server).Serve
	/opt/go/src/net/http/server.go:2795 +0x27b

API: CompleteMultipartUpload(bucket=lgtm, object=bar)
Time: 11:29:15 UTC 09/06/2018
RequestID: 1551CC189B367E86
RemoteHost: 127.0.0.1
UserAgent: aws-sdk-java/1.11.56 Linux/4.15.0-1018-gcp OpenJDK_64-Bit_Server_VM/25.181-b13/1.8.0_181
Error: open /var/lib/lgtm/minio/.minio.sys/multipart/dd938795ffd18f3ebfbf285addb1f54c59067212356d7468c464f496cc775975/f7a5f49d-4ae2-4318-aae5-1ce75a05caf9/00004.2f282b84e7e608d5852449ed940bfc51: no such file or directory
       uploadID=f7a5f49d-4ae2-4318-aae5-1ce75a05caf9, partPath=/var/lib/lgtm/minio/.minio.sys/multipart/dd938795ffd18f3ebfbf285addb1f54c59067212356d7468c464f496cc775975/f7a5f49d-4ae2-4318-aae5-1ce75a05caf9/00004.2f282b84e7e608d5852449ed940bfc51, filepath=/var/lib/lgtm/minio/.minio.sys/tmp/1412ec12-17ba-42d6-a5ce-f944c210409c/f7a5f49d-4ae2-4318-aae5-1ce75a05caf9.9e38be05-03f3-46c6-bb48-c4db957f12cf
       1: cmd/logger/logger.go:294:logger.LogIf()
       2: cmd/fs-v1-multipart.go:118:cmd.(*FSObjects).backgroundAppend()
       3: cmd/fs-v1-multipart.go:562:cmd.(*FSObjects).CompleteMultipartUpload()
       4: cmd/object-handlers.go:1702:cmd.(ObjectLayer).CompleteMultipartUpload-fm()
       5: cmd/object-handlers.go:1706:cmd.objectAPIHandlers.CompleteMultipartUploadHandler()
       6: cmd/api-router.go:58:cmd.(objectAPIHandlers).CompleteMultipartUploadHandler-fm()
       7: net/http/server.go:1947:http.HandlerFunc.ServeHTTP()
API: CompleteMultipartUpload(bucket=lgtm, object=bar)
Time: 11:29:15 UTC 09/06/2018
RequestID: 1551CC189B367E86
RemoteHost: 127.0.0.1
UserAgent: aws-sdk-java/1.11.56 Linux/4.15.0-1018-gcp OpenJDK_64-Bit_Server_VM/25.181-b13/1.8.0_181
Error: open /var/lib/lgtm/minio/.minio.sys/multipart/dd938795ffd18f3ebfbf285addb1f54c59067212356d7468c464f496cc775975/f7a5f49d-4ae2-4318-aae5-1ce75a05caf9/00002.2f282b84e7e608d5852449ed940bfc51: no such file or directory
       uploadID=f7a5f49d-4ae2-4318-aae5-1ce75a05caf9, partPath=/var/lib/lgtm/minio/.minio.sys/multipart/dd938795ffd18f3ebfbf285addb1f54c59067212356d7468c464f496cc775975/f7a5f49d-4ae2-4318-aae5-1ce75a05caf9/00004.2f282b84e7e608d5852449ed940bfc51, filepath=/var/lib/lgtm/minio/.minio.sys/tmp/1412ec12-17ba-42d6-a5ce-f944c210409c/f7a5f49d-4ae2-4318-aae5-1ce75a05caf9.9e38be05-03f3-46c6-bb48-c4db957f12cf
       1: cmd/logger/logger.go:294:logger.LogIf()
       2: cmd/fs-v1-multipart.go:595:cmd.(*FSObjects).CompleteMultipartUpload()
       3: cmd/object-handlers.go:1702:cmd.(ObjectLayer).CompleteMultipartUpload-fm()
       4: cmd/object-handlers.go:1706:cmd.objectAPIHandlers.CompleteMultipartUploadHandler()
       5: cmd/api-router.go:58:cmd.(objectAPIHandlers).CompleteMultipartUploadHandler-fm()
       6: net/http/server.go:1947:http.HandlerFunc.ServeHTTP()
```

Despite this failure, the copy of the object does actually appear to exist.

## Steps to Reproduce (for bugs)
1. Create a large file with `fallocate -l 20G /tmp/20g.bin`.
2. Upload the large file with the S3 Java client. `client.putObject(new PutObjectRequest("bucket", "file", new File("/tmp/20g.bin"));`
3. Attempt to copy the file to a new location using the `TransferManager` from the Java S3 SDK. `transferManager.copy("bucket", "file", "bucket", "someotherfile").waitForCopyResult();`
4. Observe the crash when the transfer manager attempts to complete the operation.

## Context
We've discovered that our HTTP connection to Minio times out when attempting to do a copy of a large file. To work around this we decided to switch to the multipart copy API and ran in to the described issue.

## Regression
Not as far as I know. I experienced it in `RELEASE.2018-07-23T18-34-49Z` and it continued after updating to the latest version.

## Your Environment
* Version used (`minio version`): `2018-09-01T00:38:25Z`
* Operating System and version (`uname -a`): `Linux large-file-test 4.15.0-1018-gcp #19-Ubuntu SMP Thu Aug 16 13:38:55 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux`

--- a/cmd/fs-v1-multipart.go
+++ b/cmd/fs-v1-multipart.go
@@ -621,12 +621,12 @@ func (fs *FSObjects) CompleteMultipartUpload(ctx context.Context, bucket string,
 	}
 
 	if appendFallback {
-		fsRemoveFile(ctx, file.filePath)
+		if file != nil {
+			fsRemoveFile(ctx, file.filePath)
+		}
 		for _, part := range parts {
 			partPath := getPartFile(entries, part.PartNumber, part.ETag)
-			partPath = pathJoin(uploadIDDir, partPath)
-			err = mioutil.AppendFile(appendFilePath, partPath)
-			if err != nil {
+			if err = mioutil.AppendFile(appendFilePath, pathJoin(uploadIDDir, partPath)); err != nil {
 				logger.LogIf(ctx, err)
 				return oi, toObjectErr(err)
 			}

